### PR TITLE
server(ccl)/ui: identify zone configs by zone name, not ID

### LINKED ISSUE #27718: ui: data distribution screen missing some zone configs
**Repro**

1. Follow the steps in the [Movr README](https://github.com/cockroachdb/movr) to set up a partitioned cluster, including setting up 9 zone configs (3 tables x 3 regions)
2. Look at the data distribution page

**Expected**

See all 9 zone configs you just set up

**Actual**

Only see three of the zone configs you just set up (3 tables x 1 region):

image

### PR DESCRIPTION
**Pull request by vilterp**
Previously, the `DataDistribution` endpoint was returning a map of zone configs by zone config ID. However, zone config ID is not a unique identier within `crdb_internal.zones`, since subzones attached to partitions share the same ID (that of the top-level zone config), so map entries were being overwritten and the endpoint was returning an incomplete list.

Zone names should be unique; this change uses those instead and updates the UI accordingly.

This change also introduces a `serverccl` package for the unit test to live in, since testing this code path requires creating partitions, a CCL feature.

Fixes #27718

Release note: None

### CODE REVIEW

File: pkg/server/serverpb/admin.proto [hunk 1]
```diff
@@ -563,8 +563,8 @@ message DataDistributionResponse {
   // By database name.
   map<string, DatabaseInfo> database_info = 1 [(gogoproto.nullable) = false];
 
-  // By zone config id.
-  map<int64, ZoneConfig> zone_configs = 2 [(gogoproto.nullable) = false];
```
Comment (vilterp) [inline pkg/server/serverpb/admin.proto]: Can reserve this field number and make a new field if folks think that's important.

**Review by ridwanmsharif — COMMENTED** [review]
Looks good to me, I think `zone_name` is a unique identifier for a zone, however I'm not completely sure. I defer to @knz or @benesch on that.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31407#-:-LOu331d-ZSPJN2ZbxOM:b-4bvvf0)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/server/admin.go, line 1540 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31407#-LOu2QfgC_KBw2OyBHUm:-LOu2QfgC_KBw2OyBHUn:b-ghiok) ([raw file](https://github.com/cockroachdb/cockroach/blob/b3310488e8720b7c16ca8e4082e71c98d77b0f0a/pkg/server/admin.go#L1540)):*
> ```Go
> 	zoneConfigsQuery := `
> 		SELECT zone_name, config_sql, config_protobuf 
>     FROM crdb_internal.zones WHERE cli_specifier IS NOT NULL
> ```

`WHERE zone_name IS NOT NULL`

**Review by vilterp — COMMENTED** [review]
Hm, linter seems to be mad about just having tests in a package:

```
go build github.com/cockroachdb/cockroach/pkg/ccl/serverccl: no non-test Go files in /Users/vilterp/go/src/github.com/cockroachdb/cockroach/pkg/ccl/serverccl
```

May need to move it to `partitionccl` unless there's some flag to make the linter ok with this; haven't found it yet.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31407#-:-LOuEoN54egzLDgcUhLv:b-mdopfw)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/server/admin.go, line 1540 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31407#-LOu2QfgC_KBw2OyBHUm:-LOuEt2P9nKf95YLc5vn:b-povd30) ([raw file](https://github.com/cockroachdb/cockroach/blob/b3310488e8720b7c16ca8e4082e71c98d77b0f0a/pkg/server/admin.go#L1540)):*
<details><summary><i>Previously, ridwanmsharif (Ridwan Sharif) wrote…</i></summary><blockquote>

`WHERE zone_name IS NOT NULL`
</blockquote></details>

When would it be null?

**Review by benesch — APPROVED** [review]
:lgtm:

> May need to move it to partitionccl unless there's some flag to make the linter ok with this; haven't found it yet.

Heh, convention is to just create a doc.go that contains nothing but the package name:

```go
// Package serverccl houses tests that verify CCL behavior of a running CockroachDB server.
package serverccl
```

> Reviewed 5 of 6 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31407#-:-LOuLOuNDM4isnGIuVLr:b-4jutbu)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/server/admin.go, line 1540 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31407#-LOu2QfgC_KBw2OyBHUm:-LOuJhWzDOpjOXTih86b:bxq2g9o) ([raw file](https://github.com/cockroachdb/cockroach/blob/b3310488e8720b7c16ca8e4082e71c98d77b0f0a/pkg/server/admin.go#L1540)):*
<details><summary><i>Previously, vilterp (Pete Vilter) wrote…</i></summary><blockquote>

When would it be null?
</blockquote></details>

`zone_name` is an alias for `cli_specifier`—I think Ridwan is saying to just use `zone_name` consistently throughout. To actually answer your question: it's NULL when the table has been dropped but its data is still live because of the GC TTL. The table doesn't have a name at that point, so neither does the zone.
___
*[pkg/server/serverpb/admin.proto, line 567 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/31407#-LOuL8AQA7R54vYE_ThY:-LOuL8AQA7R54vYE_ThZ:b-bryr2o) ([raw file](https://github.com/cockroachdb/cockroach/blob/6356a334fbaf7ef5c6defff3a45b4279ff437ac0/pkg/server/serverpb/admin.proto#L567)):*
> ```protobuf
> 
>   // By zone name.
>   map<string, ZoneConfig> zone_configs = 2 [(gogoproto.nullable) = false];
> ```

Even though it doesn't really matter because the admin server/client are versioned together, I do think it's good practice to bump the field number whenever its semantics change, if for no other reason than to form the habit.

**Review by vilterp — COMMENTED** [review]
TFTRs!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31407#-:-LOuU5cA-a9ljxfJCUrk:b-tlx7ko)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/server/admin.go, line 1540 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31407#-LOu2QfgC_KBw2OyBHUm:-LOuTsWaE_OQOLZ06wJ5:b29yy1g) ([raw file](https://github.com/cockroachdb/cockroach/blob/b3310488e8720b7c16ca8e4082e71c98d77b0f0a/pkg/server/admin.go#L1540)):*
<details><summary><i>Previously, benesch (Nikhil Benesch) wrote…</i></summary><blockquote>

`zone_name` is an alias for `cli_specifier`—I think Ridwan is saying to just use `zone_name` consistently throughout. To actually answer your question: it's NULL when the table has been dropped but its data is still live because of the GC TTL. The table doesn't have a name at that point, so neither does the zone.
</blockquote></details>

Oh right, forgot that the `WHERE` clause was already there.

Makes sense. Further down the road, I think showing zone configs for deleted data would be good. Will worry about that later though.
___
*[pkg/server/serverpb/admin.proto, line 567 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/31407#-LOuL8AQA7R54vYE_ThY:-LOuU3C-503KiogAuKIa:b1d32f1) ([raw file](https://github.com/cockroachdb/cockroach/blob/6356a334fbaf7ef5c6defff3a45b4279ff437ac0/pkg/server/serverpb/admin.proto#L567)):*
<details><summary><i>Previously, benesch (Nikhil Benesch) wrote…</i></summary><blockquote>

Even though it doesn't really matter because the admin server/client are versioned together, I do think it's good practice to bump the field number whenever its semantics change, if for no other reason than to form the habit.
</blockquote></details>

I buy that. Done.

**Review by benesch — APPROVED** [review]
> Reviewed 4 of 4 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31407)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

**Pull request merged**